### ISSUE TITLE
RFCs: add typing timing constants to capabilities

### ISSUE THREAD

**Issue by ShGKme** [nextcloud/spreed#9717]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**

After https://github.com/nextcloud/spreed/issues/9624 we have fixed timings in typing. 

But 10 seconds of throttling with 15s waiting [seems a bit long](https://github.com/nextcloud/spreed/pull/9689#pullrequestreview-1468209489). We may want to change this constant in the future.

The problem is, that desktop and mobile clients have multi-versioning. For proper working, all clients of one server should have the same timing.

**Describe the solution you'd like**

Add those constants to Talk server's capabilities.

**Describe alternatives you've considered**

Any other thoughts or ideas?

**Comment by Antreesy** [comment]
From the web client side, it's possible to get two constants and use it directly inside component and store.
Can't say for the mobile clients, though, some issues may arise which I'm not aware of.
In addition to that, we need fallback numbers of ms (for throttling and expiration), which should also be aligned between clients.

**Comment by ShGKme** [comment]
> From the web client side, it's possible to get two constants and use it directly inside component and store.

But we need to get them from somewhere. It cannot be just `const TYPING_TROTTLE = 10_000` (because of the desktop client).

> In addition to that, we need fallback numbers of ms (for throttling and expiration), which should also be aligned between clients.

I think the current values should be fallback.

**Comment by Antreesy** [comment]
> But we need to get them from somewhere. It cannot be just `const TYPING_TROTTLE = 10_000` (because of the desktop client).

Sorry, I meant 'get constants from capabilities', edited in the first comment.

**Comment by ShGKme** [comment]
> Could we maybe involve design team into discussion?

The main purpose of this issue is not to choose the new numbers for constants but to make them changeable.

**Comment by SystemKeeper** [comment]
> But 10 seconds of throttling with 15s waiting seems a bit long

Well the main problem is that it's not really 15s, but 25s as soon as you write 2 characters (first character triggers the first "startedTyping" message, the second sets a flag to send another "startedTyping" message after 10s). If I just write 1 character, the timeout is 15s. Anyway that might just be a feeling when testing and maybe not so much noticeable in real world.
In the last alignment call the idea of exposing the timeout values was discarded, btw.

**Comment by ShGKme** [comment]
> In the last alignment call the idea of exposing the timeout values was discarded, btw.

Yep, that is why I called it RFC. I remember that it was discarded, but I do not remember, why... Do you? 

And there are already two comments in the mentioned PR that the current solution looks too long.

**Comment by nickvergessen** [comment]
> And there are already two comments in the mentioned PR that the current solution looks too long.

You have a link?
Then again for most of the time people can not know, if the counter part was:

* typing all the time, send
* typed, deleted, typed, deleted, send
* typed, paused, typed, paused, send

**Comment by ShGKme** [comment]
> You have a link?

- https://github.com/nextcloud/spreed/pull/9689#pullrequestreview-1468209489
- https://github.com/nextcloud/spreed/pull/9689#issuecomment-1581482789
